go/src/koding/kites/kloud/utils/object/printer.go

```diff
@@ -0,0 +1,113 @@
+package object
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"reflect"
+	"strings"
+	"text/tabwriter"
+
+	"github.com/fatih/structs"
+)
+
+type Printer struct {
+	Tag    string
+	JSON   bool
+	Format string
+	W      io.Writer
+}
+
+func (p *Printer) Print(v interface{}) error {
+	if p.JSON {
+		return p.printJSON(v)
+	}
+	return p.printTab(v)
+}
+
+func (p *Printer) printJSON(v interface{}) error {
+	enc := json.NewEncoder(p.W)
+	enc.SetIndent("", "\t")
+
+	return enc.Encode(v)
+}
+
+var (
+	fprintf  = reflect.ValueOf(fmt.Fprintf)
+	stringer = reflect.TypeOf((*fmt.Stringer)(nil)).Elem()
+)
+
+func (p *Printer) printTab(v interface{}) error {
```

Comment:
Completely diff issue: if we also use json there, we would allow users to pass their own template - ie: http://kubernetes.io/docs/user-guide/jsonpath/ ( not saying we should implement this right now )